core/src/main/groovy/noe/ews/server/tomcat/TomcatProperties.groovy

```diff
@@ -2,30 +2,35 @@ package noe.ews.server.tomcat
 
 import noe.common.DefaultProperties
 import noe.common.utils.Library
+import noe.common.utils.Version
 
 
 /**
  *
  * @author Michal Karm Babacek <mbabacek@redhat.com>
  * @author Paul Lodge <plodge@redhat.com>
  *
+ *   Determine proper TOMCAT_MAJOR_VERSION or set it to null.
+ *   Do not throw any Exception as it broke execution of tests in mixtured JWS & EAP scenario.
+ *
  */
 class TomcatProperties {
-  private static final String EWS_VERSION
   public static final String TOMCAT_MAJOR_VERSION
   public static final String PUBLIC_IP_ADDRESS
 
   static {
 
-    EWS_VERSION = noe.common.DefaultProperties.ewsVersion()
-    TOMCAT_MAJOR_VERSION = Library.getUniversalProperty('tomcat.major.version')
-
-    if (!EWS_VERSION?.trim() && !TOMCAT_MAJOR_VERSION?.trim()) {
-      throw new IllegalArgumentException("Both EWS_VERSION and TOMCAT_MAJOR_VERSION are either null or empty")
-    }
-
-    if (EWS_VERSION?.trim() && !TOMCAT_MAJOR_VERSION?.trim()) {
-      throw new IllegalArgumentException("EWS_VERSION is set but the TOMCAT_MAJOR_VERSION is either null or empty")
+    Version ewsVersion = DefaultProperties.ewsVersion()
+    Version tomcatVersion = DefaultProperties.tomcatVersion()
+
+    if (tomcatVersion == null) {
+      if (ewsVersion?.getMajorVersion() == 5) {
+        TOMCAT_MAJOR_VERSION = "9"
```

Comment:
I have no in-depth look to this problem, but doesn't this introduce similar problem again? I mean - won't this default value cause a problem? Just a note though...

Reply:
I don't think so. It is set only for JWS 5 which has only one tomcat in version 9. So in this case is should be valid and right.

Reply:
not to leave things: how about like this instead
```groovy
TOMCAT_MAJOR_VERSION = [3: 7, 5: 9].get(ewsVersion?.getMajorVersion(), null)
```